modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2064,12 +2064,61 @@ public int compare(CodegenProperty one, CodegenProperty another) {
         return m;
     }
 
+    /**
+     * Process oneOf and anyOf models in a composed schema and adds them into
+     * their parent's discriminator map if the oneOf and anyOf models contain
+     * the required discriminator. If they don't contain the required
+     * discriminator or the discriminator is the wrong type then an error is
+     * thrown
+     * @param discriminator The CodegenDiscriminator that we will add models to
+     * @param c The ComposedSchema that contains the discriminator and oneOf/anyOf schemas
+     * @param discPropName The String that is the discriminator propertyName in the schema
+     */
+    private void addComposedMappedModels(CodegenDiscriminator discriminator, ComposedSchema c, String discPropName) {
+        ArrayList<List<Schema>> listOLists = new ArrayList<List<Schema>>();
+        listOLists.add(c.getOneOf());
+        listOLists.add(c.getAnyOf());
+        for (List<Schema> schemaList: listOLists) {
+            if (schemaList == null) {
+                continue;
+            }
+            for  (Schema sc: schemaList) {
+                String modelName = ModelUtils.getSimpleRef(sc.get$ref());
+                Schema refSchema = ModelUtils.getSchema(this.openAPI, modelName);
+                Schema discSchema = (Schema) refSchema.getProperties().get(discPropName);
+                if (discSchema != null && ModelUtils.isStringSchema(discSchema)) {
+                    discriminator.getMappedModels().add(new MappedModel(modelName, toModelName(modelName)));
+                    continue;
+                }
+                Boolean mapped = false;
+                if (ModelUtils.isComposedSchema(refSchema)) {
+                    ComposedSchema composedSchema = (ComposedSchema) refSchema;
+                    for  (Schema allOf: composedSchema.getAllOf()) {
+                        String allmodelName = ModelUtils.getSimpleRef(allOf.get$ref());
+                        refSchema = ModelUtils.getSchema(this.openAPI, allmodelName);
+                        discSchema = (Schema) refSchema.getProperties().get(discPropName);
+                        if (discSchema != null && ModelUtils.isStringSchema(discSchema)) {
+                            discriminator.getMappedModels().add(new MappedModel(modelName, toModelName(modelName)));
+                            mapped = true;
+                            break;
+                        }
+                    }
+                    if (mapped) {
```

Comment:
What about removing the `mapped` flag used as control flow and move the error message up? I think the logic around `mapped` is hard to follow, and could easily get messed up later.

Reply:
Removed the mapped variable and instead we now use a mappedModel variable.
If mappedModel == null, then we raise an error.